Fix/perps UI

Kahnchan opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Referral sharing now supports copying as a full referral URL in addition to raw referral codes.
  * Gift action button size can be customized for different UI contexts.
  * Referral URLs are generated with environment-aware endpoints (test vs prod) and per-source paths.
* **Public API**
  * Gift action and sharing APIs accept new flags to toggle URL copying behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Kahnchan committed:
feat: enhance GiftAction component with size prop for better customization

- Added a size prop to the GiftAction component to allow for different button sizes.
- Updated the PerpTickerBarMobile to utilize the new size prop, setting it to "small" for the GiftAction button.

Kahnchan committed:
feat(referral): enhance GiftAction and useReferFriends for URL sharing

- Added copyAsUrl prop to GiftAction component to enable sharing referral links as URLs.
- Updated useReferFriends hook to generate referral URLs based on the new copyAsUrl functionality.
- Enhanced referral dialog to display the generated referral link when copyAsUrl is true.
- Adjusted PerpsHeaderRight to utilize the updated GiftAction with URL sharing capability.

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/88c65d65-58e5-407e-ac6f-60eb6cf47b42?refs=215e32c1f05bb72258aed51290dc58bb87ab4057&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/88c65d65-58e5-407e-ac6f-60eb6cf47b42/license?refs=215e32c1f05bb72258aed51290dc58bb87ab4057&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

originalix commented on packages/shared/src/config/appConfig.ts:
```diff
@@ -58,6 +60,24 @@ export const WEB_APP_URL = 'https://app.onekey.so';
+  source: 'Perps' | 'Earn';
+  env?: IEndpointEnv;
+}): string {
+  const path = source === 'Perps' ? '/app/perp' : '/app/earn';
```
/app/perps 还是 /app/perp

这里不是刚改过 url 么

Kahnchan replied:
看到 houyao 刚改了，我改一下

Kahnchan committed:
fix(referral): update referral paths for consistency

- Changed the referral URL path from '/app/perp' to '/app/perps' for consistency across the application.
- Updated the appConfig to reflect the new path for the 'Perps' source.
- Adjusted the GiftAction component to ensure it aligns with the updated referral URL structure.

Kahnchan committed:
Merge branch 'x' into fix/perps_ui

Kahnchan committed:
Merge branch 'x' into fix/perps_ui
